internal/lightcone/preservation/textureofmemories/textureofmemories.go

```diff
@@ -74,19 +83,33 @@ func onAfterBeingAttacked(mod *modifier.ModifierInstance, e event.AttackEndEvent
 
 	// no shield + lc effect is off-cd
 	if !isShielded && !isOnCd {
-		state := mod.State().(State)
-		// apply shield
-		mod.Engine().AddShield(modshield, info.Shield{
-			Source:     mod.Owner(),
-			Target:     mod.Owner(),
-			BaseShield: info.ShieldMap{model.ShieldFormula_SHIELD_BY_SHIELDER_MAX_HP: state.shieldAmt},
+		// apply shield as buff with 2t duration
+		mod.Engine().AddModifier(mod.Owner(), info.Modifier{
```

Comment:
the OnAdd listener of shield buff uses `State`, so you need to pass a state into this modifier